libraries/bezier-rs/src/bezier/lookup.rs

```diff
@@ -4,39 +4,16 @@ use super::*;
 
 /// Functionality relating to looking up properties of the `Bezier` or points along the `Bezier`.
 impl Bezier {
-	/// Calculate the point on the curve based on the `t`-value provided.
-	pub(crate) fn unrestricted_parametric_evaluate(&self, t: f64) -> DVec2 {
-		// Basis code based off of pseudocode found here: <https://pomax.github.io/bezierinfo/#explanation>.
-
-		let t_squared = t * t;
-		let one_minus_t = 1. - t;
-		let squared_one_minus_t = one_minus_t * one_minus_t;
-
-		match self.handles {
-			BezierHandles::Linear => self.start.lerp(self.end, t),
-			BezierHandles::Quadratic { handle } => squared_one_minus_t * self.start + 2. * one_minus_t * t * handle + t_squared * self.end,
-			BezierHandles::Cubic { handle_start, handle_end } => {
-				let t_cubed = t_squared * t;
-				let cubed_one_minus_t = squared_one_minus_t * one_minus_t;
-				cubed_one_minus_t * self.start + 3. * squared_one_minus_t * t * handle_start + 3. * one_minus_t * t_squared * handle_end + t_cubed * self.end
-			}
-		}
-	}
-
-	/// Calculate the point along the curve that is a factor of `d` away from the start.
-	pub(crate) fn unrestricted_euclidean_evaluate(&self, d: f64, error: f64) -> DVec2 {
-		if let BezierHandles::Linear = self.handles {
-			return self.unrestricted_parametric_evaluate(d);
-		}
-
+	/// Convert a euclidean distance along the `Bezier` to a parametric `t`-value.
+	pub fn euclidean_to_parametric(&self, d: f64, error: f64) -> f64 {
```

Comment:
Does "euclidean distance" range from 0 to the arc length of the curve, or from 0 to 1? If it's 0 to 1, please call this a t-value not a d-value. "d" would imply actual distance (arc length) not a time ranging from 0 to 1, calculated in either euclidean or parametric space depending on context.

Reply:
I'm worried that naming the Euclidean argument to `t` we conflate it with the parametric concept of `t` we've been using elsewhere. You're right though that distance was the wrong name, because the variable is representation a ratio of the total distance along the curve - for that reason, I'd lean towards calling it `r`, `ratio` or `distance_ratio`.

Reply:
For this function alone, I've designated the variable as `ratio`, with an appropriate comment. Elsewhere I've used `t` for consistency as per our discussion on discord. Let me know if that work.